### ISSUE TITLE
Segmentation fault (core dumped)

### ISSUE THREAD

**Issue by mrostengrab** [VAST-AI-Research/UniRig#23]
Built in a dockerfile 

```
`FROM ubuntu:22.04

# Set environment variables
ENV DEBIAN_FRONTEND=noninteractive
ENV PYTHONUNBUFFERED=1
ENV LD_LIBRARY_PATH=/usr/local/cuda-12.4/lib64

# Install system dependencies and CUDA
RUN apt-get update && apt-get install -y \
    python3.11 \
    python3.11-dev \
    python3-pip \
    git \
    wget \
    gnupg2 \
    build-essential \
    libgl1-mesa-glx \
    libglib2.0-0 \
    libsm6 \
    libxext6 \
    libxrender-dev \
    libgl1-mesa-dev \
    gdb \
    valgrind \
    libassimp-dev \
    libglm-dev \
    libglfw3-dev \
    libglew-dev \
    && wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64/cuda-keyring_1.1-1_all.deb \
    && dpkg -i cuda-keyring_1.1-1_all.deb \
    && apt-get update \
    && apt-get install -y cuda-12-4 \
    && rm -rf /var/lib/apt/lists/* \
    && rm cuda-keyring_1.1-1_all.deb

# Set CUDA environment variables
ENV CUDA_HOME=/usr/local/cuda-12.4
ENV PATH=${CUDA_HOME}/bin:${PATH}
ENV LD_LIBRARY_PATH=${CUDA_HOME}/lib64:${LD_LIBRARY_PATH}

# Create a symbolic link for python3.11
RUN ln -s /usr/bin/python3.11 /usr/bin/python

# Set working directory
WORKDIR /app

# Copy requirements files
COPY requirements.txt .

# Upgrade pip and install Python dependencies
RUN pip install --upgrade pip
RUN pip install -U wheel
RUN pip install packaging
RUN python -m pip install torch==2.4.1 torchvision --index-url https://download.pytorch.org/whl/cu124
RUN python -m pip install -r requirements.txt
RUN python -m pip install spconv-cu124
RUN python -m pip install torch_scatter torch_cluster -f https://data.pyg.org/whl/torch-2.4.1+cu124.html
RUN python -m pip install numpy==1.26.4
RUN python -m pip install wheel
RUN python -m pip install --no-build-isolation flash_attn
RUN python -m pip install psutil

# Copy the rest of the application
COPY . .

# Set the default command
CMD ["/bin/bash"]`
```

trying to run the example

` bash launch/inference/generate_skeleton.sh --input examples/giraffe.glb --output results/giraffe_skeleton.fbx
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager, possibly rendering your system unusable. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv. Use the --root-user-action option if you know what you are doing and want to suppress this warning.
  0%|                                                                                                                                   | 0/1 [00:00<?, ?it/s]Info: Deleted 1 data-block(s)
Now processing examples/giraffe.glb...
Now processing examples/giraffe.glb...
16:38:57 | INFO: Data are loaded, start creating Blender stuff
16:38:57 | INFO: Blender create Mesh node Mesh_GiraffeMiddlePoly.003
16:38:57 | INFO: glTF import finished in 0.04s
save to: tmp/examples/giraffe
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.07it/s]
1 models processed
./launch/inference/extract.sh: line 58:   205 Segmentation fault      (core dumped) python -m src.data.extract --config=configs/data/quick_inference.yaml --require_suffix=obj,fbx,FBX,dae,glb,gltf,vrm --force_override=false --num_runs=1 --id=0 --time=2025_05_21_16_38_53 --faces_target_count=50000 --input=examples/giraffe.glb --output_dir=tmp
done
/usr/local/lib/python3.11/dist-packages/timm/models/layers/__init__.py:48: FutureWarning: Importing from timm.models.layers is deprecated, please import via timm.layers
  warnings.warn(f"Importing from {__name__} is deprecated, please import via timm.layers", FutureWarning)
Seed set to 12345
load task config: configs/task/quick_inference_skeleton_articulationxl_ar_256.yaml
load data config: configs/data/quick_inference.yaml
load transform config: configs/transform/inference_ar_transform.yaml
load tokenizer config: configs/tokenizer/tokenizer_parts_articulationxl_256.yaml
load model config: configs/model/unirig_ar_350m_1024_81920_float32.yaml
Flash Attention 2.0 only supports torch.float16 and torch.bfloat16 dtypes, but the current dype in OPTForCausalLM is torch.float32. You should run training or inference using Automatic Mixed-Precision via the `with torch.autocast(device_type='torch_device'):` decorator, or load the model with the `torch_dtype` argument. Example: `model = AutoModel.from_pretrained("openai/whisper-tiny", attn_implementation="flash_attention_2", torch_dtype=torch.float16)`
load system config: configs/system/ar_inference_articulationxl.yaml
Using bfloat16 Automatic Mixed Precision (AMP)
Using default `ModelCheckpoint`. Consider installing `litmodels` package to enable `LitModelCheckpoint` for automatic upload to the Lightning model registry.
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
/usr/local/lib/python3.11/dist-packages/lightning/pytorch/trainer/connectors/logger_connector/logger_connector.py:76: Starting from v1.9.0, `tensorboardX` has been removed as a dependency of the `lightning.pytorch` package, due to potential conflicts with other packages in the ML ecosystem. For this reason, `logger=True` will use `CSVLogger` as the default logger, unless the `tensorboard` or `tensorboardX` packages are found. Please `pip install lightning[extra]` or one of them to enable TensorBoard support by default
Restoring states from the checkpoint path at /root/.cache/huggingface/hub/models--VAST-AI--UniRig/snapshots/e53ad6237db5cea261727c5d50b67d140a4b9571/skeleton/articulation-xl_quantization_256/model.ckpt
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]
Loaded model weights from the checkpoint at /root/.cache/huggingface/hub/models--VAST-AI--UniRig/snapshots/e53ad6237db5cea261727c5d50b67d140a4b9571/skeleton/articulation-xl_quantization_256/model.ckpt
/usr/local/lib/python3.11/dist-packages/lightning/pytorch/trainer/connectors/data_connector.py:425: The 'predict_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=15` in the `DataLoader` to improve performance.
Predicting DataLoader 0:   0%|                                                                                                          | 0/1 [00:00<?, ?it/s]Not compiled with CUDA support
Predicting DataLoader 0: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 14.10it/s]
done`

**Comment by czpcf** [comment]
@mrostengrab This segmentation error is from bpy and has no effect on final results. Did you find anything wrong of the generated skeleton?

**Comment by mrostengrab** [comment]
Not sure if related, but trying to generate skeletons for some models makes the model size massive, and the skeletons for props are pretty bad. Anything i can tweak to help there?
https://imgur.com/a/AkWwEUG

**Comment by czpcf** [comment]
@mrostengrab Data is processed for suitable model inputs. If you need a skeleton with original mesh, use merge.sh: bash launch/inference/merge.sh --source <generated_skeleton.fbx> --target <your_obj_file.obj> --output <filename.fbx>.

Also, the models provided here are too hard to inference (donut with bad topology, or a gun with many adjacent parts). Since our model is trained on the dataset with many biped or quadruped, the performance degrades a lot for props.
